src/components/EmptyArea.tsx

```diff
@@ -0,0 +1,45 @@
+import { Box, Center, VStack, Text, HTMLChakraProps } from "@chakra-ui/react"
+import { ReactFragment } from "react"
+
+import { EmptyBoxImage } from "assets"
+
+export interface EmptyAreaOption extends HTMLChakraProps<"div"> {
+  isItemEmpty: boolean
```

Comment:
this should be an optional prop - this makes the API stricter without offering additional benefits. eg: if an initial call to an API returns `undefined`, the user now has to do a cast, which is additional friction for this component

Reply:
If they don't pass it, won't it default to `undefined` and hence, we can still do a type conversion under the hood?

Reply:
Hmm when would you have a case where I wont pass it in `isItemEmpty` first place ah? This JSX element is designed such that `isItemEmpty` is `true`, it would show the empty state, else it would render its `children`. I cant think of any case that we would need only show the empty state at **all** times, am I mistaken here?